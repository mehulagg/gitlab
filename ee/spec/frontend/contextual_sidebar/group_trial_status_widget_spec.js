import { shallowMount } from '@vue/test-utils';

import TrialStatusWidget from 'ee/contextual_sidebar/components/trial_status_widget.vue';

describe('TrialStatusWidget component', () => {
  let wrapper;

  const createComponent = () => {
    return shallowMount(TrialStatusWidget, {
      propsData: {
        href: 'billing/path-for/group',
        navIconImagePath: 'illustrations/golden_tanuki.svg',
        percentageComplete: 10,
        title: 'Gold Trial â€“ 27 days left',
      },
    });
  };

  beforeEach(() => {
    wrapper = createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('matches the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
