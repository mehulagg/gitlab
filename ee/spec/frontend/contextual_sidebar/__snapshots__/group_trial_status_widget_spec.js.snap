// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrialStatusWidget component matches the snapshot 1`] = `
<gl-link-stub
  href="billing/path-for/group"
  title="Gold Trial – 27 days left"
>
  <div
    class="gl-display-flex gl-flex-direction-column gl-align-items-stretch gl-w-full"
  >
    <span
      class="gl-display-flex gl-align-items-center"
    >
      <span
        class="nav-icon-container svg-container"
      >
        <img
          class="svg"
          src="illustrations/golden_tanuki.svg"
          width="16"
        />
      </span>
       
      <span
        class="nav-item-name gl-white-space-normal"
      >
        
        Gold Trial – 27 days left
      
      </span>
    </span>
     
    <span
      class="gl-display-flex gl-align-items-stretch gl-mt-3"
    >
      <gl-progress-bar-stub
        class="gl-flex-grow-1"
        value="10"
      />
    </span>
  </div>
</gl-link-stub>
`;
